src/Controller/MainController.php

```diff
@@ -39,8 +39,17 @@ public function indexAction(Request $request)
     {
         $response = new Response();
         $threads = $this->threadRepository->findAll();
+        $currentPage = $request->get('page', 1);
+        $limit = 10;
+        $offset = ($currentPage - 1) * $limit;
+        $totalItems = count($threads);
+        $totalPages = ceil($totalItems / $limit);
+        $itemsList = array_splice($threads, $offset, $limit);
```

Comment:
See above where I suggest only getting back from the DB what you need. It's much more efficient in the long run.